creator-node/src/reqLimiter.js

```diff
@@ -100,26 +100,28 @@ const getRateLimiter = ({
   })
 }
 
-const rateLimiterMiddleware = (req, res, next) => {
-  const definedMethods = endpointRateLimits[req.path]
-  if (!definedMethods) {
-    return next()
-  }
-  const definedLimits = definedMethods[req.method.toLowerCase()]
-  if (!definedLimits || !definedLimits.length) {
-    return next()
+/**
+ * Create an express router to attach the rate-limiting middleware
+ */
+const validRouteMethods = ['get', 'post', 'put', 'delete']
+const getRateLimiterMiddleware = () => {
```

Comment:
@jowlee would it be possible to have req, res here to pull off params if we wanted to create custom keys?

Reply:
You could change lines 114-118 to be 
```
  return (req, res, next) => {
    const customKey = createCustomKey(req, res)
    return getRateLimiter({
      prefix: customKey,
      expiry,
      max
    })(res, res, next)
  }
```
Is this what you're talking about w/ reference to "custom keys"?

Reply:
The only issue I see w/ this is that the rate limiter middleware is built w/ the generalized endpointRateLimits object from env vars, so we would need to standardize something w/ that format to work w/ this. 
ie. use a subset of the url params or request header or even req post body.

Reply:
@jowlee by default the rate limit is aggregated by req.ip (https://github.com/nfriedly/express-rate-limit#keygenerator). like lets say for a given route we want to rate limit by wallet or some param or post body instead of the default req.ip. could we add a custom keyGenerator function to the env var stringified object and pass in req to do this? i think because cnode has authentication we should do this by wallet since it's a perfect, built in rate limit field

Reply:
Made an update so you can now have the config look something like this
```
{
  "/artist_pick\":{
    post:[{
      "expiry":20,
      "max":5,
      "options":{
        "query":["trackId"],
        "body":["wallet"]
      }
    }]
  }
}
```
In this case the redis rate limit keygen would use the query param trackID is available and body arg wallet if available.